### ISSUE TITLE
[Blazor] Slick carousel error

### ISSUE THREAD

**Issue by Morhpeuss** [dotnet/aspnetcore#20775]
Hi,

I have a slick carousel on my page:

```
<div class="container">
		<div class="row slick-el bg-white border-radius-4 border px-md-5 pt-md-5 mx-0">
			@foreach (var category in this.Categories)
			{
				<div class="col-12 px-2">
					<article class="bg-white w-100  px-4 py-4 border-top-width-3 overflow-hidden position-relative align-self-stretch d-flex flex-column justify-content-between mb-4">
						<div class="row">
							<div class="col-12">
								<img class="img-fluid mx-auto w-100 d-table" style="max-width: 60px;" src="img/@(category.Icon + ".svg")">
							</div>

							<div class="col-12 text-center">
								<h3 class="min-height bold font-16 color-grey1 mt-3 my-4">@category.Name</h3>
								<div class="d-table mx-auto">
									<a href="kategorie/@category.Url" title="@category.Title" class="btn color-grey1 border-color-primary outline font-13 py-2 px-4 text-uppercase">@SharedLocalizer["link.get-more"].Value</a>
								</div>
							</div>
						</div>
					</article>
				</div>
			}
		</div>
	</div>
```
I have a function in a javascript file:

```
window.index = {
		init: function () {
			$('.slick-el').slick({
				infinite: true,
				arrows: false,
				slidesToShow: 5,
				slidesToScroll: 5,
				dots: true,
				autoplay: true,
				responsive: [
					{
						breakpoint: 1200,
						settings: {
							slidesToShow: 3,
							slidesToScroll: 3
						}
					},
					{
						breakpoint: 800,
						settings: {
							slidesToShow: 1,
							slidesToScroll: 1
						}
					}
				]
			});

			$('.slick-el-1').slick({
				arrows: true,
				slidesToShow: 2,
				slidesToScroll: 2,
				centerMode: true,
				responsive: [
					{
						breakpoint: 1200,
						settings: {
							slidesToShow: 1,
							slidesToScroll: 1
						}
					}
				],
				nextArrow:
					'<i class="fa slick-arrow right bg-grey2 icon-circle fa-arrow-right z2 cursor-pointer bg-hover-grey1"></i>',
				prevArrow:
					'<i class="fa slick-arrow left bg-grey2 icon-circle fa-arrow-left z2 cursor-pointer bg-hover-grey1"></i>'
			});
		}
	};
});
```

I call this javascript method in server-side method "OnAfterRenderAsync(bool firstRender)":

```
if (firstRender)
			{
				await this.JsRuntime.InvokeAsync<object>("index.init");
			}
```
I got an error: 

> blazor.server.js:8 Uncaught (in promise) TypeError: Cannot read property 'insertBefore' of null
    at s (blazor.server.js:8)
    at Object.t.createAndInsertLogicalContainer (blazor.server.js:8)
    at e.insertMarkup (blazor.server.js:8)
    at e.insertFrame (blazor.server.js:8)
    at e.applyEdits (blazor.server.js:8)
    at e.updateComponent (blazor.server.js:8)
    at Object.t.renderBatch (blazor.server.js:1)
    at e.<anonymous> (blazor.server.js:15)
    at blazor.server.js:15
    at Object.next (blazor.server.js:15)

I believe the problem could be that the slick script modifies the DOM.

**Comment by javiercn** [comment]
Thanks for contacting us @Morhpeuss 

It looks like this is a question about how to use ASP.NET Core. While we do our best to look through all the issues filed here, to get a faster response we suggest posting your questions to StackOverflow using the asp.net-core-blazor tag.

**Comment by Morhpeuss** [comment]
Hello,

@javiercn  I don't think that the problem is "how to use ASP.NET Core". I have 19 years of experience with .NET from WebForms /ASP.NET MVC to ASP.NET Core (and other technologies on .NET platform), so I think I know how to use ASP.NET Core. Stackoverflow was the first place where I was looking for a solution.

This could be a general problem. If I call the script where I initialize the slick carousel I get the error, If I remove the script's call .. no problem everything works perfectly.

Therefore I believe that the Blazor has a problem when an "external"  script modifies the DOM.

**Comment by Morhpeuss** [comment]
@javiercn  I did a small test, After render, I call very simple javascript:

`	$('.slick-el').empty();
	$('.slick-el').append($('<div>Hi this my div</div>'));`
As you can see, the script cleans my element and then add an new element. I get exactly same error:

> Uncaught (in promise) TypeError: Cannot read property 'insertBefore' of null
    at s (blazor.server.js:8)
    at Object.t.createAndInsertLogicalContainer (blazor.server.js:8)
    at e.insertMarkup (blazor.server.js:8)
    at e.insertFrame (blazor.server.js:8)
    at e.applyEdits (blazor.server.js:8)
    at e.updateComponent (blazor.server.js:8)
    at Object.t.renderBatch (blazor.server.js:1)
    at e.<anonymous> (blazor.server.js:15)
    at blazor.server.js:15
    at Object.next (blazor.server.js:15)

**Comment by Morhpeuss** [comment]
I found where is the problem.

I called all data services in method OnInitializedAsync, and then  I call the script in method OnAfterRenderAsync. But the method OnAfterRenderAsync doesn't wait for the method OnInitializedAsync. 

The problem is, that the script modifies the DOM and after the method OnInitializedAsync is finished, Blazor wants to re-render the DOM, the DOM was changed and it throws the exception.

When I moved all loading to the method OnInitialized, all works.
